cot_probing/eval.py

```diff
@@ -24,24 +24,29 @@ def get_answer_index_tokens_response(
             for i in range(len(question.choices))
         ],
     )
-    decoded_response = tokenizer.decode(response[0][prompt_len:])
+
+    #  model output includes the prompt, so we need to remove it
+    response = model_output[0][prompt_len:]
+
+    # Decode reponse and parse the model's answer
+    decoded_response = tokenizer.decode(response)
     answer_char = decoded_response[-1]
     if answer_char in ascii_uppercase:
         answer_idx = ascii_uppercase.index(answer_char)
     else:
         answer_idx = -1
-    return answer_idx, response[0].tolist(), decoded_response
+
+    return answer_idx, response.tolist(), decoded_response
 
 
 @dataclass
 class EvalQuestion:
-    tokens: list[int]
-    locs: dict[str, list[int]]
-    is_correct: bool
-    answer_char: str
+    correct_answer: str
+    question: str
+    tokenized_question: list[int]
```

Comment:
I think this should be called TokenizedQuestion and it doesn't need the string version